src/utils/AppRouteProvider.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { useContext, useState } from 'react';
+import propTypes from 'prop-types';
+import { useRouter } from 'next/router';
+import { useSelector } from 'react-redux';
+
+import DataProcessingIntercept from '../components/data-processing/DataProcessingIntercept';
+
+const AppRouterContext = React.createContext(null);
+
+const AppRouteProvider = (props) => {
+  const { children } = props;
+  const router = useRouter();
+
+  const [displayIntercept, setDisplayIntercept] = useState(true);
+  const [renderIntercept, setRenderIntercept] = useState(null);
+
+  const changedQCFilters = useSelector(
+    (state) => state.experimentSettings.processing.meta.changedQCFilters,
+  );
+
+  const availableIntercepts = {
+    DATA_PROCESSING: (nextRoute, hardNavigate) => (
+      <DataProcessingIntercept
+        onContinueNavigation={() => continueNavigation(nextRoute, hardNavigate)}
+        onCancelNavigation={() => cancelNavigation()}
+        onDismissIntercept={() => setDisplayIntercept(false)}
+      />
+    ),
+  };
+
+  const cancelNavigation = () => {};
```

Comment:
We could probably remove this and just not do anything on cancel

Reply:
hmm... yea that's true. I'll remove it